specfile/conditions.py

```diff
@@ -0,0 +1,141 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import re
+from typing import TYPE_CHECKING, List, Optional, Tuple
+
+from specfile.exceptions import RPMException
+from specfile.macros import Macros
+
+if TYPE_CHECKING:
+    from specfile.macro_definitions import MacroDefinitions
+    from specfile.specfile import Specfile
+
+
+def resolve_expression(
+    keyword: str, expression: str, context: Optional["Specfile"] = None
+) -> bool:
+    """
+    Resolves a RPM expression.
+
+    Args:
+        keyword: Condition keyword, e.g. `%if` or `%ifarch`.
+        expression: Expression string or a whitespace-delimited list
+          of arches/OSes in case keyword is a variant of `%ifarch`/`%ifos`.
+        context: `Specfile` instance that defines the context for macro expansions.
+
+    Returns:
+        Resolved expression as a boolean value.
+    """
+
+    def expand(s):
+        if context:
+            result = context.expand(
+                s, skip_parsing=getattr(expand, "skip_parsing", False)
+            )
+            # parse only once
+            expand.skip_parsing = True
+            return result
+        return Macros.expand(s)
```

Comment:
I am having trouble remembering contexts..;)

```suggestion
        if not context:
            return Macros.expand(s)
        
        result = context.expand(
             s, skip_parsing=getattr(expand, "skip_parsing", False)
        )
        # parse only once
        expand.skip_parsing = True
        return result
```